modules/dxp/apps/analytics/analytics-reports-web/test/js/components/Navigation.js

```diff
@@ -85,20 +86,24 @@ describe('Navigation', () => {
 					validAnalyticsConnection: false,
 				}}
 			>
-				<Navigation
-					author={testProps.author}
-					canonicalURL={testProps.canonicalURL}
-					endpoints={mockEndpoints}
-					languageTag={testProps.languageTag}
-					onSelectedLanguageClick={() => {}}
-					page={testProps.page}
-					pagePublishDate={testProps.pagePublishDate}
-					pageTitle={testProps.pageTitle}
+				<ChartStateContextProvider
+					publishDate={testProps.pagePublishDate}
 					timeRange={testProps.timeRange}
 					timeSpanKey={testProps.timeSpanKey}
-					timeSpanOptions={mockTimeSpanOptions}
-					viewURLs={mockViewURLs}
-				/>
+				>
+					<Navigation
+						author={testProps.author}
+						canonicalURL={testProps.canonicalURL}
+						endpoints={mockEndpoints}
+						languageTag={testProps.languageTag}
+						onSelectedLanguageClick={() => {}}
```

Comment:
One thing to bear in mind when passing in a no-op callback is that if the component you're rendering uses that prop in a `useEffect` or `useMemo` etc, it will "see" a new callback every single render, and that can lead to additional re-renders. One way to avoid this, or make your code robust against that possibly happening in the future is to hoist out a `noop` function to the top level of this module and always pass that in instead. You can use it in all the places you're passing a no-op function.

```js
const noop = () => {};
```

Reply:
Should I replace all occurrences for "() => {}" with "noop" and set it as you said outside the component? 🤔 Even in the tests? What is the magic here? How React know that we should not re-render things? Just to understand because this is a new concept to me :)

Reply:
Well, there's no magic. The point is just that _any_ prop you pass into a component could end up causing that component to re-render unnecessarily _if_ that prop is not stable over time, _and_ is used as an input into a `useEffect` dependencies array, or `useCallback`, or `useMemo` etc. This may be subtle too, because if the prop is passed down to a grandchild or a great-grandchild, the place where the prop is causing a re-render may be buried several levels down.

So, the basic pattern is, _whenever_ you see `<Component prop={value} />`, two of the questions you should have in mind if you want to maximize interoperability are: "_Can_ `value` be stable over time? If so, _what_ do I have to do to make it stable?" If you do that as a matter of standard practice, your code is more likely to be performant now and in the future (when it gets transplanted into some unforeseen context).

In this case, `() => {}` is not stable over time, because every time you render, you make a new anonymous function and `(() => {}) !== (() => {})`. But given `const noop = () => {}` at the top-level of the module, `noop === noop` and is stable, so it's a good pattern to follow in general. (You should think about similar gotchas with default values for function parameters: if you have a param `callback` with a default — `callback = () => {}`, then _that's_ going to be a different fallback callback every time somebody doesn't pass a default parameter.)

Reply:
You can also reuse $.noop if you enable jQuery in portal, much better

Reply:
![not-amused](https://user-images.githubusercontent.com/7074/100891107-39e78380-34b9-11eb-9f76-3915af65d934.gif)

Reply:
:crying_cat_face: